web/src/typing_events.js

```diff
@@ -25,7 +25,13 @@ const MAX_USERS_TO_DISPLAY_NAME = 3;
 
 function get_users_typing_for_narrow() {
     if (narrow_state.narrowed_by_topic_reply()) {
-        return typing_data.get_topic_typists(narrow_state.stream_id(), narrow_state.topic());
+        const current_stream_id = narrow_state.stream_id();
+        const current_topic = narrow_state.topic();
+        if (current_stream_id === undefined || current_topic === undefined) {
+            return [];
+        }
```

Comment:
You should just assert that these are not undefined; `narrow_state.narrowed_by_topic_reply` being true means those must have values, and writing conditionals like this for impossible cases is bad practice as it makes the code confusing -- a reader has to ask themselves "Oh, when can this happen, and what does this output mean?".